BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLibInternal.h

```diff
@@ -86,6 +86,21 @@ typedef struct {
   CPU_TASK         CpuTask[FixedPcdGet32 (PcdCpuMaxLogicalProcessorNumber)];
 } ALL_CPU_TASK;
 
+typedef struct {
+  UINT32  ApicId;
+  UINT32  SmmBase;
+} CPU_SMMBASE;
+
+typedef struct {
+  UINT32       Signature;
+  UINT8        CpuEntry;
+  UINT8        RegType;
+  UINT16       Unused;
+  UINT32       SmiEnReg;
+  UINT32       SmiEnVal;
```

Comment:
Do you use RegType, SmiEnReg, SmiEnVal at all ?  If not used,  better to remove it.

Reply:
If bootloader could provide SmiEnReg and SmiEnVal, it could make Payload more generic.
Better add a Length field or version field in order to update the structure later.
Should be put SMM_BASE_MIN_SIZE and SMM_BASE_GAP into this structure?

Reply:
Removed RegType, SmiEnReg and SmiRegVal for now. 

SmiEnReg and SmiEnVal can be decided by platform code on bootloader side right? Why should payload send that info again? Agree to add SMM_BASE_MIN_SIZE and SMM_BASE_GAP into structure later.

Reply:
Slim bootloader doesn't enable SMM. If UEFI payload want to enable it, UEFI payload need write SmiRegVal to SmiEnReg to enable SMM.

Reply:
But SBL can enable SMIs on S3 resume path, since we know that payload will relocate bases and enable SMIs on the first normal boot?

Reply:
This happens only when SBL finds data in SMM communication area. It means UEFI payload enabled SMM. If SMM is not enabled in payload, SBL should not do it in S3.

Reply:
I think it is more generic to let Payload fill in the register base, value, type (I/O or MMIO), length, and SBL can just enable it.  We can align with ACPI on the generic register address space descriptor.